File: crimenet_frontend/lib/models/hiring_post.dart
class HiringPost {
  final int? id;
  final int recruiterId;
  final double hourlyRate;
  final String caseType;
  final String overview;
  final String location;
  final DateTime? createdAt;
  final String? status;

  HiringPost({
    this.id,
    required this.recruiterId,
    required this.hourlyRate,
    required this.caseType,
    required this.overview,
    required this.location,
    this.createdAt,
    this.status,
  });

  factory HiringPost.fromJson(Map<String, dynamic> json) => HiringPost(
    id: json['id'],
    recruiterId: json['recruiterId'],
    hourlyRate: (json['hourlyRate'] as num).toDouble(),
    caseType: json['caseType'],
    overview: json['overview'],
    location: json['location'],
    createdAt:
        json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    status: json['status'],
  );

  Map<String, dynamic> toJson() => {
    if (id != null) 'id': id,
    'recruiterId': recruiterId,
    'hourlyRate': hourlyRate,
    'caseType': caseType,
    'overview': overview,
    'location': location,
  };
}

File: app/src/main/java/org/example/entity/HiringPost.java
package org.example.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
public class HiringPost {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recruiter_id")
    private User recruiter;

    private BigDecimal hourlyRate;
    private String caseType;
    private String overview;
    private String location;
    private LocalDateTime createdAt;

    @Enumerated(EnumType.STRING)
    private HiringPostStatus status;

    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public User getRecruiter() { return recruiter; }
    public void setRecruiter(User recruiter) { this.recruiter = recruiter; }
    public BigDecimal getHourlyRate() { return hourlyRate; }
    public void setHourlyRate(BigDecimal hourlyRate) { this.hourlyRate = hourlyRate; }
    public String getCaseType() { return caseType; }
    public void setCaseType(String caseType) { this.caseType = caseType; }
    public String getOverview() { return overview; }
    public void setOverview(String overview) { this.overview = overview; }
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public HiringPostStatus getStatus() { return status; }
    public void setStatus(HiringPostStatus status) { this.status = status; }
}